src/librustc_typeck/collect.rs

```diff
@@ -331,6 +331,7 @@ impl<'a, 'tcx> ItemCtxt<'a, 'tcx> {
         ast_generics: &hir::Generics,
         param_id: ast::NodeId,
         ty: Ty<'tcx>,
+        only_self_bounds: bool,
```

Comment:
Nit: I tend to avoid "untyped" `bool` parameters, because it makes the call site hard to read. I usually construct a wrapper type `only_self_bounds: OnlySelfBounds` like:

```rust
struct OnlySelfBounds(bool)
```

then the callsite can be:

```rust
icx.type_parameter_bounds_in_generics(ast_generics, param_id, ty, OnlySelfBounds(true))
```

which is rather clearer (inside, you have to do `self.only_self_bounds.0` or something.